### PR TITLE
fix many Error

### PR DESCRIPTION
**Pull request by songyoungwoon**
Hey brothers don't decide finish before a test!!!! 
It's very important thing

### CODE REVIEW

File: src/main/java/org/cnu/realcoding/repository/DogRepository.java [hunk 1]
```diff
@@ -88,14 +88,14 @@ public void putDogRPST(String name,
...
-                .and("ownerPhoneNumber").is(dog.getOwnerPhoneNumber());
+        criteria.is(name).and("ownerName").is(ownerName)
+                .and("ownerPhoneNumber").is(ownerPhoneNumber);
         Query query = new Query(criteria);
-        Update update = new Update();
-        update.set("medicalRecords", medicalRecord);
-        mongoTemplate.updateFirst(query, update, Dog.class);
+        for(String record : medicalRecord){
+            mongoTemplate.findOne(query, Dog.class).getMedicalRecords().add(record);
+        }
```
Comment (ywnam-stephen) [inline src/main/java/org/cnu/realcoding/repository/DogRepository.java:101]: You have to use 'update'

File: src/main/java/org/cnu/realcoding/service/DogManagementService.java [hunk 2]
```diff
@@ -91,13 +97,12 @@ public void putDogSRVC(String name,
...
-    public void addMedicalRecord(Dog dog, String medicalRecord) {
-        if(!dogRepository.check_Exist(dog.getName(), dog.getOwnerName(), dog.getOwnerPhoneNumber())) {
+    public void addMedicalRecord(String name, String ownerName, String ownerPhoneNumber, List<String> medicalRecord) {
+        if(!dogRepository.check_Exist(name, ownerName, ownerPhoneNumber)) {
             throw new DogNotFoundException();
         }
         else {
-            dog.getMedicalRecords().add(medicalRecord);
-            dogRepository.addMedicalRecord(dog, dog.getMedicalRecords());
+            dogRepository.addMedicalRecord(name, ownerName, ownerPhoneNumber, medicalRecord);
```
Comment (ywnam-stephen) [inline src/main/java/org/cnu/realcoding/service/DogManagementService.java:105]: Better update the list before this line imao.

**Review by ywnam-stephen — COMMENTED** [review]
check this

File: src/main/java/org/cnu/realcoding/repository/DogRepository.java (hunk 1 above)
Comment (songyoungwoon) [reply src/main/java/org/cnu/realcoding/repository/DogRepository.java:101]: nice play

Comment (ywnam-stephen) [reply src/main/java/org/cnu/realcoding/repository/DogRepository.java:101]: good

File: src/main/java/org/cnu/realcoding/service/DogManagementService.java (hunk 2 above)
Comment (ywnam-stephen) [reply src/main/java/org/cnu/realcoding/service/DogManagementService.java:105]: Well nevermind

**Review by ywnam-stephen — COMMENTED** [review]
done

**Comment by ywnam-stephen** [comment]
good to go

**Pull request merged**